servers/quarkus-tests/src/main/java/org/projectnessie/quarkus/tests/profiles/KeycloakTestResourceLifecycleManager.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.tests.profiles;
+
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import io.quarkus.runtime.configuration.ConfigurationException;
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.CredentialRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.representations.idm.RolesRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+
+/**
+ * This is a copy of, and a drop-in replacement for, the Keycloak test resource {@code
+ * io.quarkus.test.keycloak.server.KeycloakTestResourceLifecycleManager} from {@code
+ * io.quarkus:quarkus-test-keycloak-server}.
+ *
+ * <p>This class exists because the original one does not work with recent versions of Keycloak. It
```

Comment:
Does it make sense to open a Quarkus PR for it?

Reply:
I don't think so unfortunately, someone already asked: https://github.com/quarkusio/quarkus/issues/30072 and the answer was: use Dev Services instead.